### PR TITLE
Use latest Avalonia 0.9.1

### PR DESCRIPTION
**Pull request by worldbeater**
Updated the projects to use latest Avalonia and lifetimes API. Also added a new `SourceUriRaw` attached property to allow folks to write somewhat like the following directly in XAML, without bindings:
```xml
<Image Stretch="UniformToFill" 
       gif:GifImage.SourceUriRaw="resm:AvaloniaGif.Demo.Images.loader.gif"/>
```
Assuming the `loader.gif` is located inside the same assembly as `EmbeddedResource`. This can be sometimes useful, e.g. when the view models live in a separate project and know nothing about the currently used GUI framework

### CODE REVIEW

File: AvaloniaGif.Demo/MainWindow.xaml [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
-        Title="GIFs on Avalonia!" Height="480" Width="680" Background="White">
-    <DockPanel LastChildFill="True">
-        <ListBox DockPanel.Dock="Left" Width="300" SelectedItem="{Binding SelectedGif}" Items="{Binding AvailableGifs}"/>
-        <Image Stretch="None" gif:GifImage.SourceUri="{Binding SelectedGif}"/>
-    </DockPanel>
+        x:Class="AvaloniaGif.Demo.MainWindow"
+        Title="GIFs on Avalonia!" 
+        Height="480" Width="680"
+        Background="White">
+    <Grid>
```
Comment (jmacato) [inline AvaloniaGif.Demo/MainWindow.xaml]: Please simplify ColumnDefinitions like so:
`<Grid ColumnDefinitions="2*,3*" ...`

File: AvaloniaGif.Demo/MainWindowViewModel.cs [hunk 2]
```diff
@@ -1,64 +1,61 @@
...
+            AvailableGifs = new List<string>
+            {
+                "resm:AvaloniaGif.Demo.Images.laundry.gif",
+                "resm:AvaloniaGif.Demo.Images.earth.gif",
+                "resm:AvaloniaGif.Demo.Images.rainbow.gif",
+                "resm:AvaloniaGif.Demo.Images.newton-cradle.gif",
+                
+                // Great shots by Vitaly Silkin, free to use:
+                // https://dribbble.com/colder/projects/219798-Loaders
+                "resm:AvaloniaGif.Demo.Images.loader.gif",
```
Comment (jmacato) [inline AvaloniaGif.Demo/MainWindowViewModel.cs:27]: Great GIF source! Thanks for this :)

**Review by jmacato — CHANGES_REQUESTED** [review]
Thanks for the PR @worldbeater! :D Just need to correct some nits :) I haven't been able to maintain this repo, and we still need rendering changes in Avalonia as described by @kekekeks in https://github.com/AvaloniaUI/Avalonia/issues/3014 to bringout the best performance out of this library.

**Comment by worldbeater** [comment]
Thanks for taking the time to review this @jmacato! Applied the suggested tweak and added a `demo.gif` to `README.md` to give folks more context about what they are looking at. Decided to add an animated circle progress bar to a sample app I'm working on in my spare time, and your implementation just worked :+1: (haven't noticed any performance issues yet btw)

**Comment by jmacato** [comment]
@worldbeater Awesome! :D There's still some issues that i want to fix when i get some free time, @Kermalis 's issue on #2 is a high-priority and im still struggling on why that happens.

**Review by jmacato — APPROVED** [review]
LGTM!

**Pull request merged**